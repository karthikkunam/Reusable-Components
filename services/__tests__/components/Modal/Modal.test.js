import React from 'react';
import { shallow } from 'enzyme';
import CustomModel from '../../../src/components/Modal/Modal.jsx';


describe('Modal Component', () => {
  it('renders correctly with showModal', () => {
    const props = {
      showModal: true,
      modalTitle: '',
      modalBody: '',
      primaryButton:{ buttonLabel: 'Save Changes', className: 'btn-success' },
      secondaryButton:{ buttonLabel: 'Close', className: 'btn-secondary' },
      onModalClose: jest.fn(() => false),
    };
    const component = shallow(<CustomModel {...props} />);
    expect(component).toMatchSnapshot();
  });
  it('renders correctly with hideModal', () => {
    const props = {
      showModal: false,
      modalTitle: '',
      modalBody: '',
      primaryButton:{ buttonLabel: 'Save Changes', className: 'btn-success' },
      secondaryButton:{ buttonLabel: 'Close', className: 'btn-secondary' },
      onModalClose: jest.fn(() => true),
    };
    const component = shallow(<CustomModel {...props} />);
    expect(component).toMatchSnapshot();
  });
  it('Should hide modal on button click', () => {
    const props = {
      showModal: true,
      modalTitle: '',
      modalBody: '',
      primaryButton:{ buttonLabel: 'Save Changes', className: 'btn-success' },
      secondaryButton:{ buttonLabel: 'Close', className: 'btn-secondary' },
      onModalClose: jest.fn(() => false),
    };
    const component = shallow(<CustomModel {...props} />);
    component
      .find('button')
      .at(0)
      .simulate('click', {
        onClose: () => {},
      });
    expect(component).toMatchSnapshot();
  });
  it('Should render Primary and Secondary buttons', () => {
    const props = {
      showModal: true,
      modalTitle: '',
      modalBody: '',
      primaryButton:{ buttonLabel: 'Save Changes', className: 'btn-success' },
      secondaryButton:{ buttonLabel: 'Close', className: 'btn-secondary' },
      onModalClose: jest.fn(() => true),
    };
    const component = shallow(<CustomModel {...props} />);
    expect(component).toMatchSnapshot();
  });
  it('Should render modal with scrollable', () => {
    const props = {
      scrollable: true,
      showModal: true,
      modalTitle: '',
      modalBody: '',
      primaryButton:{ buttonLabel: 'Save Changes', className: 'btn-success' },
      secondaryButton:{ buttonLabel: 'Close', className: 'btn-secondary' },
      onModalClose: jest.fn(() => true),
    };
    const component = shallow(<CustomModel {...props} />);
    expect(component).toMatchSnapshot();
  });
  it('Should render modal with size', () => {
    const props = {
      size: 'xl',
      scrollable: true,
      showModal: true,
      modalTitle: '',
      modalBody: '',
      primaryButton:{ buttonLabel: 'Save Changes', className: 'btn-success' },
      secondaryButton:{ buttonLabel: 'Close', className: 'btn-secondary' },
      onModalClose: jest.fn(() => true),
    };
    const component = shallow(<CustomModel {...props} />);
    expect(component).toMatchSnapshot();
  });
  it('Should render alert correctly', () => {
    const props = {
      alert: true,
      showModal: true,
      modalTitle: '',
      modalBody: '',
      alertButton: { buttonLabel: 'Close', className: 'btn-primary btn-lg btn-block' },
      onModalClose: jest.fn(() => true),
    };
    const component = shallow(<CustomModel {...props} />);
    expect(component).toMatchSnapshot();
  });
});
