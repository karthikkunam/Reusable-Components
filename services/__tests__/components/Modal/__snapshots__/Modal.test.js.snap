// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal Component Should hide modal on button click 1`] = `
<Bootstrap(Modal)
  animation={true}
  autoFocus={false}
  enforceFocus={false}
  keyboard={false}
  onHide={[Function]}
  restoreFocus={false}
  show={true}
>
  <div
    className="modal-content "
  >
    <div
      className=""
    >
      <h5
        className="modal-header"
      >
        <span
          className="modal-title"
        />
        <button
          aria-label="Close"
          className="close"
          data-dismiss="modal"
          onClick={[Function]}
          type="button"
        >
          <span
            aria-hidden="true"
          >
            ×
          </span>
        </button>
      </h5>
      <div
        className="modal-body"
      />
      <div
        className="modal-footer"
      >
        <div
          className="modal-footer-inner btn-right"
        >
          <button
            className="btn btn-secondary"
            data-dismiss="modal"
            onClick={[Function]}
            onEnter={[Function]}
            type="button"
          >
            Close
          </button>
          <button
            className="btn btn-success"
            type="button"
          >
            Save Changes
          </button>
        </div>
      </div>
    </div>
  </div>
</Bootstrap(Modal)>
`;

exports[`Modal Component Should render Primary and Secondary buttons 1`] = `
<Bootstrap(Modal)
  animation={true}
  autoFocus={false}
  enforceFocus={false}
  keyboard={false}
  onHide={[Function]}
  restoreFocus={false}
  show={true}
>
  <div
    className="modal-content "
  >
    <div
      className=""
    >
      <h5
        className="modal-header"
      >
        <span
          className="modal-title"
        />
        <button
          aria-label="Close"
          className="close"
          data-dismiss="modal"
          onClick={[Function]}
          type="button"
        >
          <span
            aria-hidden="true"
          >
            ×
          </span>
        </button>
      </h5>
      <div
        className="modal-body"
      />
      <div
        className="modal-footer"
      >
        <div
          className="modal-footer-inner btn-right"
        >
          <button
            className="btn btn-secondary"
            data-dismiss="modal"
            onClick={[Function]}
            onEnter={[Function]}
            type="button"
          >
            Close
          </button>
          <button
            className="btn btn-success"
            type="button"
          >
            Save Changes
          </button>
        </div>
      </div>
    </div>
  </div>
</Bootstrap(Modal)>
`;

exports[`Modal Component Should render alert correctly 1`] = `
<Bootstrap(Modal)
  animation={true}
  autoFocus={false}
  enforceFocus={false}
  keyboard={false}
  onHide={[Function]}
  restoreFocus={false}
  show={true}
>
  <div
    className="modal-content "
  >
    <div
      className="alert alert-dismissible undefined"
    >
      <h5
        className="alert-heading"
      >
        <span
          className="alert-heading-text"
        />
        <span
          className="icon icon-status-info-reversed"
        />
      </h5>
      <div
        className="alert-body"
      />
      <button
        className="btn btn-primary btn-lg btn-block"
        type="button"
      >
        Close
      </button>
    </div>
  </div>
</Bootstrap(Modal)>
`;

exports[`Modal Component Should render modal with scrollable 1`] = `
<Bootstrap(Modal)
  animation={true}
  autoFocus={false}
  enforceFocus={false}
  keyboard={false}
  onHide={[Function]}
  restoreFocus={false}
  scrollable={true}
  show={true}
>
  <div
    className="modal-content "
  >
    <div
      className=""
    >
      <h5
        className="modal-header"
      >
        <span
          className="modal-title"
        />
        <button
          aria-label="Close"
          className="close"
          data-dismiss="modal"
          onClick={[Function]}
          type="button"
        >
          <span
            aria-hidden="true"
          >
            ×
          </span>
        </button>
      </h5>
      <div
        className="modal-body"
      />
      <div
        className="modal-footer"
      >
        <div
          className="modal-footer-inner btn-right"
        >
          <button
            className="btn btn-secondary"
            data-dismiss="modal"
            onClick={[Function]}
            onEnter={[Function]}
            type="button"
          >
            Close
          </button>
          <button
            className="btn btn-success"
            type="button"
          >
            Save Changes
          </button>
        </div>
      </div>
    </div>
  </div>
</Bootstrap(Modal)>
`;

exports[`Modal Component Should render modal with size 1`] = `
<Bootstrap(Modal)
  animation={true}
  autoFocus={false}
  enforceFocus={false}
  keyboard={false}
  onHide={[Function]}
  restoreFocus={false}
  scrollable={true}
  show={true}
  size="xl"
>
  <div
    className="modal-content "
  >
    <div
      className=""
    >
      <h5
        className="modal-header"
      >
        <span
          className="modal-title"
        />
        <button
          aria-label="Close"
          className="close"
          data-dismiss="modal"
          onClick={[Function]}
          type="button"
        >
          <span
            aria-hidden="true"
          >
            ×
          </span>
        </button>
      </h5>
      <div
        className="modal-body"
      />
      <div
        className="modal-footer"
      >
        <div
          className="modal-footer-inner btn-right"
        >
          <button
            className="btn btn-secondary"
            data-dismiss="modal"
            onClick={[Function]}
            onEnter={[Function]}
            type="button"
          >
            Close
          </button>
          <button
            className="btn btn-success"
            type="button"
          >
            Save Changes
          </button>
        </div>
      </div>
    </div>
  </div>
</Bootstrap(Modal)>
`;

exports[`Modal Component renders correctly with hideModal 1`] = `
<Bootstrap(Modal)
  animation={true}
  autoFocus={false}
  enforceFocus={false}
  keyboard={false}
  onHide={[Function]}
  restoreFocus={false}
  show={false}
>
  <div
    className="modal-content "
  >
    <div
      className=""
    >
      <h5
        className="modal-header"
      >
        <span
          className="modal-title"
        />
        <button
          aria-label="Close"
          className="close"
          data-dismiss="modal"
          onClick={[Function]}
          type="button"
        >
          <span
            aria-hidden="true"
          >
            ×
          </span>
        </button>
      </h5>
      <div
        className="modal-body"
      />
      <div
        className="modal-footer"
      >
        <div
          className="modal-footer-inner btn-right"
        >
          <button
            className="btn btn-secondary"
            data-dismiss="modal"
            onClick={[Function]}
            onEnter={[Function]}
            type="button"
          >
            Close
          </button>
          <button
            className="btn btn-success"
            type="button"
          >
            Save Changes
          </button>
        </div>
      </div>
    </div>
  </div>
</Bootstrap(Modal)>
`;

exports[`Modal Component renders correctly with showModal 1`] = `
<Bootstrap(Modal)
  animation={true}
  autoFocus={false}
  enforceFocus={false}
  keyboard={false}
  onHide={[Function]}
  restoreFocus={false}
  show={true}
>
  <div
    className="modal-content "
  >
    <div
      className=""
    >
      <h5
        className="modal-header"
      >
        <span
          className="modal-title"
        />
        <button
          aria-label="Close"
          className="close"
          data-dismiss="modal"
          onClick={[Function]}
          type="button"
        >
          <span
            aria-hidden="true"
          >
            ×
          </span>
        </button>
      </h5>
      <div
        className="modal-body"
      />
      <div
        className="modal-footer"
      >
        <div
          className="modal-footer-inner btn-right"
        >
          <button
            className="btn btn-secondary"
            data-dismiss="modal"
            onClick={[Function]}
            onEnter={[Function]}
            type="button"
          >
            Close
          </button>
          <button
            className="btn btn-success"
            type="button"
          >
            Save Changes
          </button>
        </div>
      </div>
    </div>
  </div>
</Bootstrap(Modal)>
`;
